sql/internet_outages/italy_covid19_outage_v1/query.sql

```diff
@@ -0,0 +1,193 @@
+DECLARE country_code STRING DEFAULT "IT";
+DECLARE fromdate DATE DEFAULT '2020-03-01'; -- '2019-12-01'
+DECLARE todate DATE DEFAULT '2020-03-02'; --'2020-03-31';
+
+-- The minimum size of a bucket for it to be published.
+DECLARE MIN_NUMBER_CLIENTS INT64 DEFAULT 5000;
+
+-- Note: udf.udf_json_extract_int_map map doesn't work in this case as it expects an INT -> INT
+-- map, while we have a STRING->int map
+CREATE TEMP FUNCTION
+  udf_json_extract_string_to_int_map (input STRING) AS (ARRAY(
+    SELECT
+      STRUCT(CAST(SPLIT(entry, ':')[OFFSET(0)] AS STRING) AS key,
+             CAST(SPLIT(entry, ':')[OFFSET(1)] AS INT64) AS value)
+    FROM
+      UNNEST(SPLIT(REPLACE(TRIM(input, '{}'), '"', ''), ',')) AS entry
+    WHERE
+      LENGTH(entry) > 0 ));
+     
+
+-- Get a stable source for DAUs.
+WITH DAUs AS (
+    SELECT
+        submission_date as date,
+        COUNT(*) as client_count
+    FROM
+        telemetry.clients_daily
+    WHERE
+        submission_date >= fromdate
+        AND submission_date <= todate
+        AND country = country_code
+    GROUP BY 1
+),
+-- Compute aggregates for the health data.
+health_data_sample AS (
+    SELECT 
+        DATE(SAFE_CAST(creation_date as TIMESTAMP)) as date,
```

Comment:
The `submission_date` in the WHERE clause is required because of the partitioning of the dataset. It's also handy for us, since filtering by `submission_date` actually makes this dataset reproducible: if we only filtered by creation date, then we'd have to add some "slack time" since pings submitted later on might have a creation date way earlier than that. This is especially true for outages.

Reply:
Also, added fun: `telemetry.clients_daily` doesn't allow to filter by `creation_date`, which makes normalizing a bit weird, but generally ok.

Reply:
In general, it would not help us: we're interested in when the outage was _detected_ (i.e. when the ping was created) versus when the outage was _reported_ (i.e. submission time). This would basically shift the "spikes" for certain classes of outages to dates in the future.

For this specific case it doesn't make much difference, but I'd like to keep this code as close as possible to what we use for other cases. Unless you have a strong opinion on this, of course!